utils/schema/add-datasource-ids.ts

```diff
@@ -0,0 +1,84 @@
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+
+import type { DataSourceConfig } from '../types/DataSourceConfig';
+
+import { getAllDataSourceFiles } from '../lib/DataSource';
+
+const ARTIFACT_SCHEMA_PATH = './artifact.json';
+const CORE_DATASOURCE_IDS_PATH = './core-datasource-ids.json';
+const DATASOURCE_BASE_PATH = '../../';
+
+// TODO: infer the type from the schema?
```

Comment:
Is this TODO still valid, or something we'll address at a later point?

Reply:
Oh good point, I suppose that is up for discussion. Since Typescript is _structurally_ typed, and because we only care about the type defined here, then I'm not sure if it's worth generating the type for the entire schema.

What do you think? If you agree, I can remove the `TODO`

Reply:
I care more about the artifact being structurally correct, so im fine to leave it as is and remove the TODO :)